### PR TITLE
[Merged by Bors] - feat: port Algebra.Lie.CartanMatrix

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Lie/CartanMatrix.lean [hunk 1]
```diff
@@ -66,23 +66,24 @@ exceptional semisimple Lie algebras.
...
+  * `CartanMatrix.E₆`
+  * `CartanMatrix.E₇`
+  * `CartanMatrix.E₈`
+  * `CartanMatrix.F₄`
+  * `CartanMatrix.G₂`
+  * `LieAlgebra.E₆`
+  * `LieAlgebra.E₇`
+  * `LieAlgebra.E₈`
+  * `LieAlgebra.F₄`
+  * `LieAlgebra.G₂`
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/CartanMatrix.lean:79]: I'm very strongly of the opinion that these should remain lowercase. It is standard practice in mathematics to use:

- uppercase for the cartan matrix
- lowercase for the Lie algebra
- uppercase for the Lie group (but they aren't in mathlib yet)

File: Mathlib/Algebra/Lie/CartanMatrix.lean [hunk 2]
```diff
@@ -163,15 +164,15 @@ def adF : B × B → FreeLieAlgebra R (Generators B) :=
 
 private theorem ad_E_of_eq_eq_zero (i : B) (h : A i i = 2) : adE R A ⟨i, i⟩ = 0 := by
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/CartanMatrix.lean:165]:
```suggestion
private theorem adE_of_eq_eq_zero (i : B) (h : A i i = 2) : adE R A ⟨i, i⟩ = 0 := by
```

File: Mathlib/Algebra/Lie/CartanMatrix.lean [hunk 3]
```diff
@@ -163,15 +164,15 @@ def adF : B × B → FreeLieAlgebra R (Generators B) :=
...
   have h' : (-2 : ℤ).toNat = 0 := by rfl
-  simp [ad_E, h, h']
+  simp [adE, h, h']
 
 private theorem ad_F_of_eq_eq_zero (i : B) (h : A i i = 2) : adF R A ⟨i, i⟩ = 0 := by
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/CartanMatrix.lean:169]:
```suggestion
private theorem adF_of_eq_eq_zero (i : B) (h : A i i = 2) : adF R A ⟨i, i⟩ = 0 := by
```

File: Mathlib/Algebra/Lie/CartanMatrix.lean [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+-- mathport name: exprE
+local notation "E" => FreeLieAlgebra.of R ∘ Generators.E
+
+-- mathport name: exprF
+local notation "F" => FreeLieAlgebra.of R ∘ Generators.F
+
+-- mathport name: exprad
+local notation "ad" => LieAlgebra.ad R (FreeLieAlgebra R (Generators B))
+
+/-- The terms correpsonding to the `⁅H, H⁆`-relations. -/
```
Comment (jcommelin) [inline Mathlib/Algebra/Lie/CartanMatrix.lean:129]: While we're tweaking this file, could you please to a global replace:

`correpsonding` ↦ `corresponding`

**Comment by jcommelin** [comment]
bors d+

File: Mathlib/Algebra/Lie/CartanMatrix.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/Algebra/Lie/CartanMatrix.lean]: Oops, I didn't realize I'd done this. Fixed.

**Pull request closed without merging**